src/fast-path.js

```diff
@@ -343,6 +343,21 @@ FastPath.prototype.needsParens = function() {
           return false;
       }
 
+    case "TSParenthesizedType": {
+      if (
+        parent.type === "VariableDeclarator" ||
+        parent.type === "TypeAnnotation" ||
+        parent.type === "GenericTypeAnnotation"
+      ) {
+        return false;
+      }
+      // Delegate to inner TSParenthesizedType
+      if (node.typeAnnotation.type === "TSParenthesizedType") {
+        return false;
```

Comment:
On a mobile connection right now, so I'll start with vscode... (Haven't pulled down the TS/Angular repos on my macbook since it was formatted 😢 )